### PR TITLE
libva: 2.4.1 -> 2.5.0

### PR DESCRIPTION
**Pull request by colemickens**
###### Motivation for this change
Update to latest `libva`.

I'm currently rebuilding most of the world to test it.

cc: @worldofpeace

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by colemickens** [comment]
This fails for me, due to `libva-vdpau-driver`. I don't even know what is pulling this, but my system includes it, I guess:

```
these derivations will be built:
  /nix/store/lnd57cpk1k3f6fnw9xrw9djs02c455ha-libva-vdpau-driver-0.7.4.drv
building '/nix/store/lnd57cpk1k3f6fnw9xrw9djs02c455ha-libva-vdpau-driver-0.7.4.drv'...
unpacking sources
unpacking source archive /nix/store/blj6mqwhbcpwmg3pffj7h58yhvnsfdnj-libva-vdpau-driver-0.7.4.tar.bz2
source root is libva-vdpau-driver-0.7.4
setting SOURCE_DATE_EPOCH to timestamp 1349449403 of file libva-vdpau-driver-0.7.4/debian.upstream/libva-vdpau-driver.install
patching sources
applying patch /nix/store/k6wrzfkvnjjr5gk2daq98jsmws68v46x-glext85.patch
patching file src/utils_glx.h
applying patch /nix/store/shkcn3jnz2jgc94x4rjww190jkf59wxd-libva-vdpau-driver-0.7.4-VAEncH264VUIBufferType.patch
patching file src/vdpau_dump.c
configuring
patching file src/vdpau_decode.c
fixing libtool script ./ltmain.sh
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/5hg66rw48822n8pyq3akz08y5yinzb88-libva-vdpau-driver-0.7.4
checking for a BSD-compatible install... /nix/store/k8lhqzpaaymshchz8ky3z4653h4kln9d-coreutils-8.31/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /nix/store/k8lhqzpaaymshchz8ky3z4653h4kln9d-coreutils-8.31/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking how to print strings... printf
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking dependency style of gcc... none
checking for a sed that does not truncate output... /nix/store/vnyd3wh5i5kj66n9c5b8shzxjjrw22cn-gnused-4.7/bin/sed
checking for grep that handles long lines and -e... /nix/store/agcay3wmf74qinwshnjqy73w8rxf82hs-gnugrep-3.3/bin/grep
checking for egrep... /nix/store/agcay3wmf74qinwshnjqy73w8rxf82hs-gnugrep-3.3/bin/grep -E
checking for fgrep... /nix/store/agcay3wmf74qinwshnjqy73w8rxf82hs-gnugrep-3.3/bin/grep -F
checking for ld used by gcc... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... no
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse nm output from gcc object... ok
checking for sysroot... no
./configure: line 1: /usr/bin/file: not found
checking for mt... no
checking if : is a manifest tool... no
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (ld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking dependency style of gcc... (cached) none
checking whether byte ordering is bigendian... no
checking for ANSI C header files... (cached) yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking whether make supports nested variables... yes
checking for pthread_create in -lpthread... yes
checking for tan in -lm... yes
checking for timer_create in -lrt... yes
checking for clock_gettime... yes
checking whether __attribute__((visibility())) is supported... yes
checking for GL/gl.h... yes
checking for GL/glext.h... yes
checking for GL/glx.h... yes
checking for glXCreateContext in -lGL... yes
checking for pkg-config... /nix/store/01a3dppfi3rv03m8yx2nwxbkklggyzf9-pkg-config-0.29.2/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for LIBVA_DEPS... yes
checking for LIBVA_X11_DEPS... yes
checking for LIBVA_GLX_DEPS... yes
checking for VA drivers path... /nix/store/5hg66rw48822n8pyq3akz08y5yinzb88-libva-vdpau-driver-0.7.4/lib/dri
checking for VDPAU... yes
checking for VDPAU/MPEG-4 support... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating debian.upstream/Makefile
config.status: creating src/Makefile
config.status: creating src/config.h
config.status: executing depfiles commands
config.status: executing libtool commands

libva-vdpau-driver configuration summary:

VA-API version ................... : 1.5.0
VA-API drivers path .............. : /nix/store/5hg66rw48822n8pyq3akz08y5yinzb88-libva-vdpau-driver-0.7.4/lib/dri
VDPAU version .................... :
VDPAU/MPEG-4 support ............. : yes
GLX support ...................... : yes

building
build flags: SHELL=/nix/store/xfghy8ixrhz3kyy6p724iv3cxji088dx-bash-4.4-p23/bin/bash
Making all in debian.upstream
make[1]: Entering directory '/build/libva-vdpau-driver-0.7.4/debian.upstream'
sed \
  -e 's|\@PACKAGE_VERSION\@|0.7.4|' \
  -e 's|\@LIBVA_PACKAGE_VERSION\@|1.0.9|' \
  -e 's|\@LIBVA_DRIVERS_PATH\@|/nix/store/5hg66rw48822n8pyq3akz08y5yinzb88-libva-vdpau-driver-0.7.4/lib/dri|' \
  -e 's|\@DATE\@|'"`LC_ALL=C date +'%a, %d %b %Y %X %z'`"'|' \
  changelog.in > changelog
sed \
  -e 's|\@PACKAGE_VERSION\@|0.7.4|' \
  -e 's|\@LIBVA_PACKAGE_VERSION\@|1.0.9|' \
  -e 's|\@LIBVA_DRIVERS_PATH\@|/nix/store/5hg66rw48822n8pyq3akz08y5yinzb88-libva-vdpau-driver-0.7.4/lib/dri|' \
  -e 's|\@DATE\@|'"`LC_ALL=C date +'%a, %d %b %Y %X %z'`"'|' \
  control.in > control
sed \
  -e 's|\@PACKAGE_VERSION\@|0.7.4|' \
  -e 's|\@LIBVA_PACKAGE_VERSION\@|1.0.9|' \
  -e 's|\@LIBVA_DRIVERS_PATH\@|/nix/store/5hg66rw48822n8pyq3akz08y5yinzb88-libva-vdpau-driver-0.7.4/lib/dri|' \
  -e 's|\@DATE\@|'"`LC_ALL=C date +'%a, %d %b %Y %X %z'`"'|' \
  libva-vdpau-driver.install.in > libva-vdpau-driver.install
make[1]: Leaving directory '/build/libva-vdpau-driver-0.7.4/debian.upstream'
Making all in src
make[1]: Entering directory '/build/libva-vdpau-driver-0.7.4/src'
make  all-am
make[2]: Entering directory '/build/libva-vdpau-driver-0.7.4/src'
  CC     debug.lo
  CC     object_heap.lo
  CC     uasyncqueue.lo
  CC     ulist.lo
  CC     uqueue.lo
  CC     utils.lo
  CC     vdpau_buffer.lo
In file included from vdpau_buffer.c:24:0:
vdpau_video.h:278:12: warning: 'struct v4l2_buffer' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_buffer *v4l2_buf,
            ^~~~~~~~~~~
vdpau_video.h:277:12: warning: 'struct v4l2_format' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_format *v4l2_fmt,
            ^~~~~~~~~~~
  CC     vdpau_decode.lo
In file included from vdpau_decode.c:25:0:
vdpau_video.h:278:12: warning: 'struct v4l2_buffer' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_buffer *v4l2_buf,
            ^~~~~~~~~~~
vdpau_video.h:277:12: warning: 'struct v4l2_format' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_format *v4l2_fmt,
            ^~~~~~~~~~~
vdpau_decode.c: In function 'get_VdpDecoderProfile':
vdpau_decode.c:78:5: warning: 'VAProfileH264Baseline' is deprecated [-Wdeprecated-declarations]
     case VAProfileH264Baseline: return VDP_DECODER_PROFILE_H264_BASELINE;
     ^~~~
In file included from /nix/store/q3lx5rhv4s4k7630xx0dp72158pfclyv-libva-2.5.0-dev/include/va/va_backend.h:32:0,
                 from vdpau_driver.h:24,
                 from vdpau_decode.h:24,
                 from vdpau_decode.c:22:
/nix/store/q3lx5rhv4s4k7630xx0dp72158pfclyv-libva-2.5.0-dev/include/va/va.h:369:5: note: declared here
     VAProfileH264Baseline va_deprecated_enum = 5,
     ^~~~~~~~~~~~~~~~~~~~~
vdpau_decode.c: In function 'vdpau_QueryConfigProfiles':
vdpau_decode.c:1061:9: warning: 'VAProfileH264Baseline' is deprecated [-Wdeprecated-declarations]
         VAProfileH264Baseline,
         ^~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/q3lx5rhv4s4k7630xx0dp72158pfclyv-libva-2.5.0-dev/include/va/va_backend.h:32:0,
                 from vdpau_driver.h:24,
                 from vdpau_decode.h:24,
                 from vdpau_decode.c:22:
/nix/store/q3lx5rhv4s4k7630xx0dp72158pfclyv-libva-2.5.0-dev/include/va/va.h:369:5: note: declared here
     VAProfileH264Baseline va_deprecated_enum = 5,
     ^~~~~~~~~~~~~~~~~~~~~
vdpau_decode.c: In function 'vdpau_QueryConfigEntrypoints':
vdpau_decode.c:1111:5: warning: 'VAProfileH264Baseline' is deprecated [-Wdeprecated-declarations]
     case VAProfileH264Baseline:
     ^~~~
In file included from /nix/store/q3lx5rhv4s4k7630xx0dp72158pfclyv-libva-2.5.0-dev/include/va/va_backend.h:32:0,
                 from vdpau_driver.h:24,
                 from vdpau_decode.h:24,
                 from vdpau_decode.c:22:
/nix/store/q3lx5rhv4s4k7630xx0dp72158pfclyv-libva-2.5.0-dev/include/va/va.h:369:5: note: declared here
     VAProfileH264Baseline va_deprecated_enum = 5,
     ^~~~~~~~~~~~~~~~~~~~~
  CC     vdpau_driver.lo
In file included from vdpau_subpic.h:24:0,
                 from vdpau_driver.c:27:
vdpau_video.h:278:12: warning: 'struct v4l2_buffer' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_buffer *v4l2_buf,
            ^~~~~~~~~~~
vdpau_video.h:277:12: warning: 'struct v4l2_format' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_format *v4l2_fmt,
            ^~~~~~~~~~~
  CC     vdpau_dump.lo
  CC     vdpau_gate.lo
In file included from vdpau_gate.c:23:0:
vdpau_video.h:278:12: warning: 'struct v4l2_buffer' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_buffer *v4l2_buf,
            ^~~~~~~~~~~
vdpau_video.h:277:12: warning: 'struct v4l2_format' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_format *v4l2_fmt,
            ^~~~~~~~~~~
  CC     vdpau_image.lo
In file included from vdpau_image.c:23:0:
vdpau_video.h:278:12: warning: 'struct v4l2_buffer' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_buffer *v4l2_buf,
            ^~~~~~~~~~~
vdpau_video.h:277:12: warning: 'struct v4l2_format' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_format *v4l2_fmt,
            ^~~~~~~~~~~
  CC     vdpau_mixer.lo
In file included from vdpau_mixer.c:23:0:
vdpau_video.h:278:12: warning: 'struct v4l2_buffer' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_buffer *v4l2_buf,
            ^~~~~~~~~~~
vdpau_video.h:277:12: warning: 'struct v4l2_format' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_format *v4l2_fmt,
            ^~~~~~~~~~~
  CC     vdpau_subpic.lo
In file included from vdpau_subpic.h:24:0,
                 from vdpau_subpic.c:22:
vdpau_video.h:278:12: warning: 'struct v4l2_buffer' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_buffer *v4l2_buf,
            ^~~~~~~~~~~
vdpau_video.h:277:12: warning: 'struct v4l2_format' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_format *v4l2_fmt,
            ^~~~~~~~~~~
  CC     vdpau_video.lo
In file included from vdpau_video.c:22:0:
vdpau_video.h:278:12: warning: 'struct v4l2_buffer' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_buffer *v4l2_buf,
            ^~~~~~~~~~~
vdpau_video.h:277:12: warning: 'struct v4l2_format' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_format *v4l2_fmt,
            ^~~~~~~~~~~
vdpau_video.c:931:12: warning: 'struct v4l2_buffer' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_buffer *v4l2_buf,
            ^~~~~~~~~~~
vdpau_video.c:930:12: warning: 'struct v4l2_format' declared inside parameter list will not be visible outside of this definition or declaration
     struct v4l2_format *v4l2_fmt,
            ^~~~~~~~~~~
vdpau_video.c:927:1: error: conflicting types for 'vdpau_CreateSurfaceFromV4L2Buf'
 vdpau_CreateSurfaceFromV4L2Buf(
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from vdpau_video.c:22:0:
vdpau_video.h:274:1: note: previous declaration of 'vdpau_CreateSurfaceFromV4L2Buf' was here
 vdpau_CreateSurfaceFromV4L2Buf(
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:468: vdpau_video.lo] Error 1
make[2]: Leaving directory '/build/libva-vdpau-driver-0.7.4/src'
make[1]: *** [Makefile:340: all] Error 2
make[1]: Leaving directory '/build/libva-vdpau-driver-0.7.4/src'
make: *** [Makefile:308: all-recursive] Error 1
builder for '/nix/store/lnd57cpk1k3f6fnw9xrw9djs02c455ha-libva-vdpau-driver-0.7.4.drv' failed with exit code 2
error: build of '/nix/store/lnd57cpk1k3f6fnw9xrw9djs02c455ha-libva-vdpau-driver-0.7.4.drv' failed

```

We might need to import new(er) patches from Gentoo? https://packages.gentoo.org/packages/x11-libs/libva-vdpau-driver

**Comment by worldofpeace** [comment]
patchsets in lots of different places
* https://git.archlinux.org/svntogit/packages.git/tree/trunk?h=packages/libva-vdpau-driver
* https://src.fedoraproject.org/rpms/libva-vdpau-driver/tree/master

**Comment by colemickens** [comment]
Okay, I've rebuilt my system with this and #65396 and things work well. Doesn't fix `vaapi+iris` but I didn't expect it to. This should be good to merge to `staging`.

File: pkgs/development/libraries/vaapi-vdpau/default.nix [hunk 1]
```diff
@@ -13,23 +8,25 @@ stdenv.mkDerivation rec {
     sha256 = "1fcvgshzyc50yb8qqm6v6wn23ghimay23ci0p8sm8gxcy211jp0m";
   };
 
-  patches = [ ./glext85.patch
-              (fetchurl { url = "http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/x11-libs/libva-vdpau-driver/files/libva-vdpau-driver-0.7.4-VAEncH264VUIBufferType.patch?revision=1.1";
-                          name = "libva-vdpau-driver-0.7.4-VAEncH264VUIBufferType.patch";
-                          sha256 = "166svcav6axkrlb3i4rbf6dkwjnqdf69xw339az1f5yabj72pqqs";
-                        }) ];
+  patches = [
+    ./libva-vdpau-driver-0.7.4-glext-85.patch
```
Comment (worldofpeace) [inline pkgs/development/libraries/vaapi-vdpau/default.nix]: Was it not possible to fetch these patches from anywhere?

Comment (worldofpeace) [inline pkgs/development/libraries/vaapi-vdpau/default.nix]: Was it not possible to fetch these patches from anywhere?

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Commit msg should not have `v`s in it.

File: pkgs/development/libraries/vaapi-vdpau/default.nix (hunk 1 above)
Comment (colemickens) [reply pkgs/development/libraries/vaapi-vdpau/default.nix]: To be honest, I don't understand the preference to fetch the patches from elsewhere. FWIW, I pulled all of the patches from Fedora except the last one, from Gentoo.

Comment (colemickens) [reply pkgs/development/libraries/vaapi-vdpau/default.nix]: Of course, they were being fetched before, so I can update it if necessary....

**Comment by colemickens** [comment]
I dropped the 'v's from the commit.

File: pkgs/development/libraries/vaapi-vdpau/default.nix (hunk 1 above)
Comment (worldofpeace) [reply pkgs/development/libraries/vaapi-vdpau/default.nix]: I think the problem it's a problem that comes with nixpkgs being a monorepo.
In fedora it's a separate repo for each package so it's fine, but for nixpkgs we'll make the repo size simply huge for everyone.

Comment (colemickens) [reply pkgs/development/libraries/vaapi-vdpau/default.nix]: Okay, changed to `fetchpatch` and pushed.

**Pull request merged**